### PR TITLE
Bug 1164355 - Migrate app modal dialog to smart-modal-dialog

### PR DESCRIPTION
**Pull request by shamenchens**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7c0f42a923c54b04cd0fc96cb5e567dc7b1ae516

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=854d25496e8052691fe7bdbeb511b2dd088eb824

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cedfdc6edb19cca0181288142d573480296b75a3

File: tv_apps/smart-system/js/app_modal_dialog.js [hunk 1]
```diff
@@ -63,6 +63,14 @@
   };
 
   AppModalDialog.prototype._fetchElements = function amd__fetchElements() {
+    // XXX: Place another container outside of the original element to prevent
```
Comment (huchengtw-moz) [inline tv_apps/smart-system/js/app_modal_dialog.js]: Please file a bug for refactoring smart-modal-dialog to support DOM element and put the bug here.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=99d9c9f1b5f5527d3ff6d0b6838f6238f40f0a39

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a75f656c8bf0e9a10ee6d082e942baa7eb00be3e

**Pull request merged**